### PR TITLE
feat: create tables for HEPData

### PR DESCRIPTION
**Pull request by AndriiPovsten**
The possible HEPData submission of Statistical Inference figures using `hepdata.py` which is located in utils folder.

### CODE REVIEW

**Comment by AndriiPovsten** [comment]
I'm not sure about the `submission.yaml` file since the user need to specify what exactly they want to see as an UI.
And also, is it necessary to upload figures to the `hed_data` folder. Because this requires adding syntax to the `submission_hep_data` function (theoretically user can upload drawings to the folder manually)?

File: .DS_Store
Comment (alexander-held) [inline .DS_Store:1]: These files should also be removed.

File: analyses/cms-open-data-ttbar/cabinetry_config.yml
Comment (alexander-held) [inline analyses/cms-open-data-ttbar/cabinetry_config.yml:1]: A few files here change whether they have a newline at the end of the file or not, please also avoid these changes to minimize the files touched by this pull request if the changes are not needed.

Comment (oshadura) [reply analyses/cms-open-data-ttbar/cabinetry_config.yml:1]: @AndriiPovsten please check why these files were touched (did you add an empty line at the end?), and please fix it.

**Review by oshadura — COMMENTED** [review]
@AndriiPovsten there are some linter warnings, please check and I think it is ready to go. Another question, did you add the option `ENABLE_HEPDATA` in the notebook so actually we can run it as a part of demo tomorrow?

**Comment by oshadura** [comment]
Ops, I was commenting not up-to-date version of the pull request, I will go again through the pull request.

File: .DS_Store
Comment (alexander-held) [inline .DS_Store:1]: one more file here to be removed

File: analyses/cms-open-data-ttbar/utils/hepdata.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    table_name = ""
+    if independent_variables:
+        table_name = f"4j{index}b Figure"
+    elif independent_variables_ml:
+        table_name = f"Feature{index} Figure"
+
+    table = Table(table_name)
+
+    var = Variable("sample", is_independent=True, is_binned=False, units="should be some units for the samples??")
```
Comment (alexander-held) [inline analyses/cms-open-data-ttbar/utils/hepdata.py:48]: The units depend on the distribution, for HT / masses they should be GeV, the delta R distributions are unitless.

**Review by oshadura — APPROVED** [review]
LGTM!

**Comment by oshadura** [comment]
@alexander-held your opinion?

File: analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py [hunk 2]
```diff
@@ -723,13 +723,21 @@ def get_query(source):
...
 # %%
 if USE_INFERENCE:
-    utils.plotting.plot_data_mc(model_prediction, model_prediction_postfit, data_ml, config_ml)
+    utils.plotting.plot_data_mc(model_prediction_ml, model_prediction_postfit_ml, data_ml, config_ml)
+
+# %%
+if utils.config["preservation"]["HEPData"] is True:
+    import utils.hepdata
+    #Submission of model prediction
+    utils.hepdata.submission_hep_data(model, model_prediction, "hepdata_model", cabinetry_config)
```
Comment (alexander-held) [inline analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py:738]: Perhaps a small renaming here would be useful: the function does not submit, but it prepares the material for submission.

File: analyses/cms-open-data-ttbar/README.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ This directory is focused on running the CMS Open Data $t\bar{t}$ analysis throu
 | models/                       | Contains models used for ML inference task (when `USE_TRITON = False`)                                                                          |
 | utils/                        | Contains code for bookkeeping and cosmetics, as well as some boilerplate. Also contains images used in notebooks.                               |
 | utils/config.py               | This is a general config file to handle different options for running the analysis.                               |
+| utils/hepdata.py              | This is the .py file with function which would create a tables which would be submitted and stored into the [HEP_DATA website](https://www.hepdata.net) (use `HEP_DATA = True`)     |
```
Comment (alexander-held) [inline analyses/cms-open-data-ttbar/README.md:21]:
```suggestion
| utils/hepdata.py              | Function to create tables for submission to the [HEP_DATA website](https://www.hepdata.net) (use `HEP_DATA = True`)     |
```

File: analyses/cms-open-data-ttbar/README.md [hunk 4]
```diff
@@ -51,3 +52,36 @@ argument is the appropriate reference file for the number of files per process a
 For full usage help see the output of `python validate_histograms.py --help`.
 
 `validate_histograms.py` can also be used to create new references by passing the `--dump-json` option.
+
+#### HEP data creation and submision.
+For proper submission, you need to modify the `submission.yaml` with proper explanation of variables and your table.
+To submit the created histograms to HEP data,, you'll need to install the necessary packages and make some modifications to `ttbar_analysis_pipeline.ipynb` notebook.
```
Comment (alexander-held) [inline analyses/cms-open-data-ttbar/README.md:58]:
```suggestion
To submit the created histograms to HEP data, you'll need to install the necessary packages and make some modifications to the `ttbar_analysis_pipeline.ipynb` notebook.
```

File: analyses/cms-open-data-ttbar/README.md [hunk 5]
```diff
@@ -51,3 +52,36 @@ argument is the appropriate reference file for the number of files per process a
...
+```
+
+```python
+from getpass import getpass
+import os
+# Get the password securely
+password = getpass("Enter your password: ")
+
+command = f"hepdata-cli upload '/home/cms-jovyan/analysis-grand-challenge/analyses/cms-open-data-ttbar/hepdata_model.zip.zip' -e yourname.yoursurname@cern.ch"
+os.system(f'echo {password} | {command}') #insert your passport in the actived window
```
Comment (alexander-held) [inline analyses/cms-open-data-ttbar/README.md:84]:
```suggestion
os.system(f'echo {password} | {command}') #insert your password in the active window
```

**Comment by alexander-held** [comment]
resolves #108

**Pull request merged**